### PR TITLE
DNN: IE Backend: Not work with other width-height

### PR DESCRIPTION
**Pull request by allnes**
Issue: [https://github.com/opencv/opencv/issues/13724](https://github.com/opencv/opencv/issues/13724)

### CODE REVIEW

File: modules/dnn/src/layers/prior_box_layer.cpp [hunk 1]
```diff
@@ -487,9 +487,7 @@ class PriorBoxLayerImpl CV_FINAL : public PriorBoxLayer
         if (_explicitSizes)
         {
             InferenceEngine::Builder::PriorBoxClusteredLayer ieLayer(name);
-
-            CV_Assert(_stepX == _stepY);
-            ieLayer.setStep(_stepX);
+            ieLayer.setSteps({_stepX, _stepY});
```
Comment (dkurt) [inline modules/dnn/src/layers/prior_box_layer.cpp]: According to it's implementation, we should pass `{_stepY, _stepX}` instead:

https://github.com/opencv/dldt/blob/17e66dc5a6631d630da454506902bd7c25d4170b/inference-engine/src/inference_engine/builders/ie_prior_box_clustered_layer.cpp#L83-L89

File: modules/dnn/src/layers/prior_box_layer.cpp [hunk 2]
```diff
@@ -529,6 +526,12 @@ class PriorBoxLayerImpl CV_FINAL : public PriorBoxLayer
             if (!_aspectRatios.empty())
             {
                 l.getParameters()["aspect_ratio"] = _aspectRatios;
+                if (_stepX != _stepY)
+                {
+                    l.getParameters()["step"] = 0.0;
+                    l.getParameters()["step_h"] = _stepY;
+                    l.getParameters()["step_w"] = _stepX;
+                }
```
Comment (dkurt) [inline modules/dnn/src/layers/prior_box_layer.cpp]: Why this parameters under `if (!_aspectRatios.empty())` condition but not in the `else` section of `if (_stepX == _stepY)`?

File: samples/dnn/object_detection.cpp [hunk 3]
```diff
@@ -153,51 +153,39 @@ void postprocess(Mat& frame, const std::vector<Mat>& outs, Net& net)
     std::vector<int> classIds;
     std::vector<float> confidences;
     std::vector<Rect> boxes;
-    if (net.getLayer(0)->outputNameToIndex("im_info") != -1)  // Faster-RCNN or R-FCN
+    if (outLayerType == "DetectionOutput")
```
Comment (dkurt) [inline samples/dnn/object_detection.cpp]: Please rebase a branch. These changes already merged.

Could you also add some test which covers this bug?

File: modules/dnn/test/test_ie_models.cpp [hunk 4]
```diff
@@ -259,5 +259,46 @@ INSTANTIATE_TEST_CASE_P(/**/,
     Combine(testing::ValuesIn(getAvailableTargets(DNN_BACKEND_INFERENCE_ENGINE)), intelModels())
 );
 
+typedef TestWithParam< tuple< Size, Target > > IEBackend_Test;
+TEST_P(IEBackend_Test, Frame_Width_Height_Test)
```
Comment (dkurt) [inline modules/dnn/test/test_ie_models.cpp]: I think that it'd be better to replace this test to [test_backends.cpp](https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_backends.cpp).

**Comment by allnes** [comment]
@dkurt done fix

File: modules/dnn/test/test_backends.cpp [hunk 5]
```diff
@@ -162,6 +162,18 @@ TEST_P(DNNTestNetwork, MobileNet_SSD_Caffe)
                inp, "detection_out", "", diffScores);
 }
 
+TEST_P(DNNTestNetwork, MobileNet_SSD_Caffe_Different_Width_Height)
```
Comment (dkurt) [inline modules/dnn/test/test_backends.cpp]: This test is failed for me locally with `DNN_TARGET_MYRIAD`. May I ask you to take a look? The rest of test are OK.

**Pull request merged**